test/apm-client/http-apm-client/config.test.js

```diff
@@ -567,10 +567,6 @@ test('payloadLogFile', function (t) {
             `expected line ${n} in the log file to match item no ${n} received by the server`,
           );
         });
-
-        file.on('end', function () {
-          t.end();
-        });
```

Comment:
@david-luna Since this has been merged, I'm seeing flaky test failures in that test file *much more often*. See https://github.com/elastic/apm-agent-nodejs/issues/3313#issuecomment-1734291517
The flaky test failure is always with node v20 or v21 (a nightly build test failure).  I haven't yet reproduced it locally, which makes it harder.

Reply:
@david-luna In https://github.com/elastic/apm-agent-nodejs/issues/3313#issuecomment-1666227024 you said:

> Changed to a sync file read and loop to avoid it in https://github.com/elastic/apm-agent-nodejs/pull/3532

However, that is not yet your patch here is doing. Did you have a different change that didn't make this PR?